### PR TITLE
initial autodocs attempt

### PR DESCRIPTION
**Pull request by jreidinger**
auto deployment of documentation added add http://yast-ui-bindings.surge.sh/

in future nicer initial markdown would be nice and of course improve a bit documentation itself would be nice.

### CODE REVIEW

File: .surge.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+
+set -x
+
+if [ "$TRAVIS_EVENT_TYPE" = "push" -a "$TRAVIS_BRANCH" = "master" ]; then
+  docker run -it -e TRAVIS=1 -e TRAVIS_JOB_ID="$TRAVIS_JOB_ID" -e SURGE_LOGIN="$SURGE_LOGIN" -e SURGE_TOKEN="$SURGE_TOKEN" yast-ycp-ui-bindings-image /bin/bash -c "zypper  in -y npm doxygen; npm install --global surge; rake doc; surge --project ./autodoc/html --domain yast-ui-bindings.surge.sh"
```
Comment (lslezak) [inline .surge.sh]: `ELINETOOLONG`

File: .travis.yml [hunk 2]
```diff
@@ -9,3 +9,8 @@ script:
   # the "yast-travis-cpp" script is included in the base yastdevel/cpp image
   # see https://github.com/yast/docker-yast-cpp/blob/master/yast-travis-cpp
   - docker run -it -e TRAVIS=1 -e TRAVIS_JOB_ID="$TRAVIS_JOB_ID" yast-ycp-ui-bindings-image yast-travis-cpp
+
+after_success:
+  - echo $TRAVIS_EVENT_TYPE
```
Comment (lslezak) [inline .travis.yml]: This is just for debugging?

File: .surge.sh (hunk 1 above)
Comment (lslezak) [inline .surge.sh]: NP: Also think about forks, if you enable Travis for your fork it won't work.

- By default `$SURGE_LOGIN` `$SURGE_TOKEN` will be undefined, resulting in invalid credentials
- Even if you provide your credentials it won't work as the `yast-ui-bindings.surge.sh` domain is blocked for the original account.

You should probably enable the deployment only in the original `yast/yast-ycp-ui-bindings` GitHub repo. (See [how to check the Github repo](https://github.com/yast/yast.github.io/blob/master/.travis_deploy.rb#L37).)

File: Rakefile [hunk 3]
```diff
@@ -4,3 +4,10 @@ Yast::Tasks.configuration do |conf|
   #lets ignore license check for now
   conf.skip_license_check << /.*/
 end
+
+task :doc do
+  rm_rf "autodoc"
+  mkdir "autodoc"
```
Comment (mvidner) [inline Rakefile]: In all the other repositories, the name is "autodoc**s**". Please rename it here.

File: .travis.yml (hunk 2 above)
Comment (jreidinger) [inline .travis.yml]: yes, I plan to keep it there for some kind of verified that it always send only from our repo and master

Comment (jreidinger) [inline .travis.yml]: but if statement is evaluated, so I can remove it

File: .surge.sh (hunk 1 above)
Comment (jreidinger) [inline .surge.sh]: well, checking forks are probably too much for this simple script. And btw if after_success fails, then travis job do not fail.

File: .surge.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+
```
Comment (mvidner) [inline .surge.sh]: Please add a comment stating what the purpose of this script is, and some link to help people not familiar with the service. My guess:
> Build the documentation and publish it on a public site
> It uses Mordor, https://mordor.mil/how_to_join

File: .surge.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+set -x
+
+if [ "$TRAVIS_EVENT_TYPE" = "push" -a "$TRAVIS_BRANCH" = "master" ]; then
```
Comment (mvidner) [inline .surge.sh]: This feels wrong. Why would I check Travis variables in a Surge script? The Travis checks should be done in the Travis script.

File: .surge.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  docker run -it -e TRAVIS=1 -e TRAVIS_JOB_ID="$TRAVIS_JOB_ID" \
+    -e SURGE_LOGIN="$SURGE_LOGIN" -e SURGE_TOKEN="$SURGE_TOKEN" \
```
Comment (mvidner) [inline .surge.sh]: I guess this will fail if I run it without some kind of a setup. Please explain in a comment.

File: .surge.sh (hunk 5 above)
Comment (jreidinger) [inline .surge.sh]: problem is that travis script do not have shell to execute, so it failed ( already tried ).

File: .surge.sh (hunk 6 above)
Comment (jreidinger) [inline .surge.sh]: ok, it will mainly fail, because you do not have permission to that surge page. otherwise simple `surge` will work in interactive mode.

**Pull request merged**